src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java

```diff
@@ -1076,7 +1076,7 @@ public String getDigitalSignatureAlgorithm() {
      * {@inheritDoc}
      */
     public int getDigitalSignatureKeyLength() {
-        return getESAPIProperty(DIGITAL_SIGNATURE_KEY_LENGTH, 1024);
+        return getESAPIProperty(DIGITAL_SIGNATURE_KEY_LENGTH, 2048);
```

Comment:
Worth asking why we don't just default to 4096?

Reply:
We probably should switch to using EdDSA or ECDSA if EdDSA is not supported in the standard JDK 8, but for now (and until 2030), I think 2048-bits is considered fine by NIST and I wasn't sure of the performance impact of bumping it to 4096 (it wouldn't be noticeable unless someone was doing a lot of them in a tight loop).  So, I figure if they want it better, let them tweak ESAPI.properties. That's one reason that even though I'd been aware of this for several years, I wasn't overly concerned about it. I pretty much only did it because of a GitHub issue someone opened.

Maybe not a great answer, but at this point, I don't really want to go back and change it.